include/features.php

```diff
@@ -16,14 +16,10 @@
 function feature_enabled($uid, $feature) {
 	$x = Config::get('feature_lock', $feature, false);
 
-	if (!$x) {
-		$x = PConfig::get($uid, 'feature', $feature, false);
-		if (!$x) {
-			$x = Config::get('feature', $feature, false);
-			if (!$x) {
-				$x = get_feature_default($feature);
-			}
-		}
+	if ($x === false) {
+		$x = get_feature_default($feature);
+		$x = Config::get('feature', $feature, $x);
+		$x = PConfig::get($uid, 'feature', $feature, $x);
```

Comment:
This always performs at least 3 SQL queries while the previous structure perfomed at most 3 SQL queries, why the change?

Also, line 17 could have been included in the flattening as well, becoming the last call after line 22.

Reply:
The change was a part of a longer debug experience :-) I thought it would look nicer.

Reply:
It does indeed, at the price performance, and I wondered why you didn’t go all the way.

Reply:
Yeah, it seems that you are right.

Reply:
I cannot include the line 17 in it, since this line overrides the other lines. BTW: I don't think that this is a huge performance, since we are caching the result anyway.

What do you think?

Reply:
If you still want to cache it without unnecessary calls, you can do it this way:
````php
$x = Config::get('feature_lock', $feature, false);
if ($x === false) {
    $x = PConfig::get($uid, 'feature', $feature, false);
}
if ($x === false) {
    $x = Config::get('feature', $feature, false);
}
if ($x === false) {
    $x = get_feature_default($feature);
}
````